### PR TITLE
FOGL-7128: Search empty readings table before creating new database i…

### PR DESCRIPTION
**Pull request by gnandan**
…f reading tables are not available

Signed-off-by: nandan <nandan.ghildiyal@gmail.com>

### CODE REVIEW

**Comment by praveen-garg** [comment]
@gnandan This needs conflict resolution. @MarkRiddoch can you please review this?

**Comment by MarkRiddoch** [comment]
We are about to do another release, so, unfortunately I would suggest we hold this until after that release.

File: C/common/config_category.cpp [hunk 1]
```diff
@@ -4,7 +4,7 @@
  * Copyright (c) 2018 Dianomic Systems
  *
  * Released under the Apache 2.0 Licence
- *
+
```
Comment (MarkRiddoch) [inline C/common/config_category.cpp:7]: Please revert this file back to what it was

File: C/plugins/north/OMF/plugin.cpp [hunk 2]
```diff
@@ -2,6 +2,7 @@
  * Fledge PI Server north plugin.
  *
  * Copyright (c) 2018-2022 Dianomic Systems
+
```
Comment (MarkRiddoch) [inline C/plugins/north/OMF/plugin.cpp:5]: Please revert this file back to what it was

**Review by MarkRiddoch — CHANGES_REQUESTED** [review]
Just a few minor things. Slightly concerned as to thread safety, but nothing to indicate it isn't. Have all the system tests been run against this branch? I think we should before merging this to develop.

**Review by MarkRiddoch — APPROVED** [review]

File: C/plugins/storage/sqlite/common/readings_catalogue.cpp [hunk 3]
```diff
@@ -1960,52 +1973,73 @@ ReadingsCatalogue::tyReadingReference  ReadingsCatalogue::getReadingReference(Co
...
+	if (m_EmptyAssetReadingCatalogue.size() == 0)
+	{
+		loadEmptyAssetReadingCatalogue();
+	}
+	
+	auto it = m_EmptyAssetReadingCatalogue.begin();
+	if (it != m_EmptyAssetReadingCatalogue.end())
+	{
+		asset = it->first;
+		emptyTableReference.tableId = it->second.first;
```
Comment (Innovative-ashwin) [inline C/plugins/storage/sqlite/common/readings_catalogue.cpp:2037]: if there will be only one value in the map , and iteration over a set of values not require d, then what is the need of map data structure?

Comment (gnandan) [reply C/plugins/storage/sqlite/common/readings_catalogue.cpp:2037]: Purge task keeps purging data on a regular interval, that's why map is being used to track of all the empty table.

**Review by Innovative-ashwin — APPROVED** [review]

**Pull request merged**